### PR TITLE
Add templates

### PR DESCRIPTION
**Pull request by kprosise**
Templates added for tutorials, user guides and reference manual pages. These can be found in `templates/`.

QA steps were not performed.

This commit applies to Jira FFTK Epic #1792
This commit applies to Jira FFTK Story #1794
This commit applies to Jira tasks #1906, #1907, #1908, #1909

Signed-off-by: Katrina Prosise <katrina.prosise@foundries.io>

## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

Added templates.
This can be especially useful for new contributors, as well as for adding new pages.
The aim is to provide a general structure, but not "hard rules" that must be followed.
More detailed information about different section pages can be found in the style guide.

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [x] Run spelling and grammar check, preferably with linter.
* [x] Avoid changing any header associated with a link/reference.
* [x] Step through instructions (or ask someone to do so).
* [x] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [x] Match tone and style of page/section.
* [x] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [x] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [x] follow best practices for commits.
  * [x] Descriptive title written in the imperative.
  * [x] Include brief overview of QA steps taken.
  * [x] Mention any related issues numbers.
  * [x] End message with sign off/DCO line (`-s, --signoff`).
  * [x] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [x] Request PR review by a technical writer and at least one peer.

## Comments

Feel free to make any suggestions, or you can make commits later — "field testing" will likely show were they can improve.

### CODE REVIEW

**Comment by doanac** [comment]
Docs for d4ca2d74d3c2b42f03dee96c47b5efe68c7351ea are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1793/docs/artifacts/html/index.html

File: templates/reference-manual.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    Prints help message
+
+Example Usage (optional)
+------------------------
+
+Keep it very simple, more showing than explaining.
+You can have a sentence or two of what the example achieves.
+For example, a fully configured ``local.conf``,
+or a CLI tool showing  a common use-case of the tool.
+A walk-through/tutorial/how-to belongs in the user-guide.
```
Comment (angolini) [inline templates/reference-manual.rst:53]: A Reference Manual describes the tools from a toolbox.
A User Guide describes how to use one tool from the toolbox.

I'm not sure where this belongs, but I like that definition.

**Review by angolini — APPROVED** [review]
I feel like a "how to add an image" is missing. This is the one thing that I always need to search again how to perform

File: templates/reference-manual.rst (hunk 1 above)
Comment (kprosise) [reply templates/reference-manual.rst:53]: That definition is great! I will find a place for it, likely in the style guide or templates. Thanks for sharing.

**Comment by kprosise** [comment]
> I feel like a "how to add an image" is missing. This is the one thing that I always need to search again how to perform

I think putting a Sphinx and RsT cheat-sheet here in `docs` is a good idea. There's also various extensions that could use a bit of an explanation. I'll create a task for this.

File: templates/reference-manual.rst (hunk 1 above)
Comment (kprosise) [inline templates/reference-manual.rst:53]:
```suggestion
A walk-through/tutorial/how-to belongs in the user-guide.
"A Reference Manual describes the tools from a toolbox,
a User Guide describes how to use one tool from the toolbox."
```

**Comment by doanac** [comment]
Docs for d118c58494af50f7f5d0e9590f82d00e55226464 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1796/docs/artifacts/html/index.html

**Pull request merged**